packages/ag-charts-enterprise/src/features/error-bar/errorBar.ts

```diff
@@ -88,14 +99,52 @@ export class ErrorBars
         this.cartesianSeries = ctx.series as _ModuleSupport.CartesianSeries<any, any>;
         const { contentGroup } = this.cartesianSeries;
 
-        this.groupNode = new _Scene.Group({ name: `${contentGroup.id}-series-errorBars` });
+        this.groupNode = new _Scene.Group({
+            name: `${contentGroup.id}-series-errorBars`,
+            zIndex: _ModuleSupport.Layers.SERIES_ERRORBAR_ZINDEX,
+            zIndexSubOrder: this.cartesianSeries.getGroupZIndexSubOrder('error-bars'),
+        });
         contentGroup.appendChild(this.groupNode);
         this.selection = _Scene.Selection.select(this.groupNode, () => this.errorBarFactory());
 
-        this.destroyFns.push(this.cartesianSeries.addListener('data-model', (event) => this.onDataProcessed(event)));
+        this.destroyFns.push(
+            this.cartesianSeries.addListener(
+                'processData-prerequest',
+                (event: _ModuleSupport.SeriesPrerequestDataEvent) => this.onPrerequestData(event)
+            )
+        );
+        this.destroyFns.push(
+            this.cartesianSeries.addListener('data-update', (event: _ModuleSupport.SeriesDataUpdateEvent) =>
+                this.onDataUpdate(event)
+            )
+        );
+        this.destroyFns.push(
+            this.cartesianSeries.addListener('visibility-changed', (event: _ModuleSupport.SeriesVisibilityEvent) =>
+                this.onToggleSeriesItem(event)
+            )
+        );
     }
 
-    onDataProcessed(event: {
+    onPrerequestData(event: {
+        props: _ModuleSupport.PropertyDefinition<string>[];
+        isContinuousX: boolean;
+        isContinuousY: boolean;
+    }) {
+        const { cartesianSeries, xLowerKey, xUpperKey, yLowerKey, yUpperKey } = this;
+        const { isContinuousX, isContinuousY } = event;
+        event.props.push(
+            valueProperty(cartesianSeries, yLowerKey, isContinuousY, { id: 'yValue' }),
+            valueProperty(cartesianSeries, yUpperKey, isContinuousY, { id: 'yValue' })
+        );
+        if (xLowerKey !== undefined && xUpperKey !== undefined) {
+            event.props.push(
+                valueProperty(cartesianSeries, xLowerKey, isContinuousX, { id: 'xValue' }),
+                valueProperty(cartesianSeries, xUpperKey, isContinuousX, { id: 'xValue' })
+            );
+        }
```

Comment:
I would prefer if we treated the event as immutable, and instead returned additional props to append to the data-model properties, as it's a bit easier to reason about?

Reply:
I agree; I thought about this too initially. It's will be a bit trickier to implement because currently `Listeners.dispatch` returns `void`, but I'll see whether it's feasible to change this.

Reply:
Resolved in [5e1b3d780](https://github.com/ag-grid/ag-charts/commit/5e1b3d780a7dc1be647bc7937d2085bd99ca1424)